### PR TITLE
KMS Support

### PR DESCRIPTION
**Pull request by kyleknap**
This adds support for KMS. Note that any input argument that is modeled as a blob must be in bytes (it gets validated before making the request).

cc @danielgtaylor @jamesls

### CODE REVIEW

File: boto/kms/layer1.py [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+        :param encryption_context: Name:value pair that specifies the
+            encryption context to be used for authenticated encryption. For
+            more information, see `Authenticated Encryption`_.
+
+        :type grant_tokens: list
+        :param grant_tokens: A list of grant tokens that represent grants which
+            can be used to provide long term permissions to perform encryption.
+
+        """
+        if plaintext is not None:
```
Comment (jamesls) [inline boto/kms/layer1.py]: Is there a reason for the `is not None` check?  `plaintext` is a required param.

File: boto/kms/layer1.py [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+            if not isinstance(plaintext, six.binary_type):
+                raise TypeError(
+                    "Argument ``plaintext`` is "
+                    "modeled as a blob. Value must be bytes.")
```
Comment (jamesls) [inline boto/kms/layer1.py]: I suspect users may not understand the `modeled as a blob` bit.  What if we just say something like "value must be of type %s" % six.binary_type?

**Comment by jamesls** [comment]
Looks good, just a question about the is not None check for required params.

**Comment by kyleknap** [comment]
@jamesls I updated the code with your suggestions. Let me know what you think.

**Comment by jamesls** [comment]
:shipit: Looks good.

**Pull request merged**